flink-connectors/flink-connector-base/src/main/java/org/apache/flink/connector/base/source/hybrid/HybridSourceSplitEnumerator.java

```diff
@@ -224,7 +226,9 @@ public void handleSourceEvent(int subtaskId, SourceEvent sourceEvent) {
             }
 
             // track readers that have finished processing for current enumerator
+            // TODO: should finishedReaders be reset after switching to a new numerator?
```

Comment:
@tweise It seems that `finishedReaders` will keep increasing after the first switch unless the job is restarted. So if there are more than 2 sources in the chain, the 3rd and latter source may never get triggered. Is my understand correct?